pkg/healthcheck/healthcheck.go

```diff
@@ -932,3 +960,39 @@ func validateDataPlanePodReporting(pods []*pb.Pod) error {
 
 	return nil
 }
+
+func checkUnschedulablePods(pods []corev1.Pod) error {
+	var podName []string
+	for _, pod := range pods {
+		for _, condition := range pod.Status.Conditions {
+			if condition.Reason == "Unschedulable" {
+				podName = append(podName, pod.Name)
+			}
+		}
+	}
+
+	if len(podName) != 0 {
+		pods := strings.Join(podName, ",")
+		return fmt.Errorf("No node available for the pods %s", pods)
+	}
+
+	return nil
+}
+
+func checkPodSecurityPolicies(rst []v1beta1.ReplicaSet) error {
+	var replicaSets []string
+	for _, rs := range rst {
+		for _, r := range rs.Status.Conditions {
+			if strings.Contains(r.Message, "unable to validate against any pod security policy") {
+				replicaSets = append(replicaSets, rs.Name)
+			}
+		}
+	}
+
+	if len(replicaSets) != 0 {
+		replicasets := strings.Join(replicaSets, ",")
+		return fmt.Errorf("Invalid pod security policy for the replicasets %s", replicasets)
```

Comment:
How about `Pod security policy does not allow creation of pods: %s`?

Reply:
Something like `Pod security policy does not allow creation of pods for the repliasets %s` ?